chat-server/server.js

```diff
@@ -16,11 +23,47 @@ const welcomeMessage = {
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
-
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+app.get("/messages", function (request, response) {
+  response.json({ messages });
+});
+app.get("/messages/latest", function (request, response) {
+  const latestMessages = messages.slice(0, 10);
+  response.send(latestMessages);
+});
+app.get("/messages/:id", function (request, response) {
+  let messageId = Number(request.params.id);
+  const filteredMessages = messages.filter(
+    (message) => message.id === messageId
+  );
+  response.send(filteredMessages);
+});
+app.post("/messages", function (request, response) {
+  const timeUTC = new Date().toLocaleTimeString();
+  let newMessage = {
+    id: messages.length,
+    from: request.body.from,
+    text: request.body.text,
+    timeSent: timeUTC,
+  };
+  console.log(newMessage);
+  if (request.body.from && request.body.text) {
+    messages.push(newMessage);
+    response.json(messages);
+  } else response.status(400).send("Please check the fields have been correctly filled in");
```

Comment:
I've been testing this under https://andrius-chat-serverr.onrender.com/messages and it's giving me this error even when I put the text and from and I can see it from the body.

Reply:
fixed, now it works :)